src/hnsw/build.c

```diff
@@ -434,8 +434,19 @@ static void BuildIndex(
         opts.expansion_search = metadata.expansion_search;
         opts.metric_kind = metadata.metric_kind;
     } else {
-        BlockNumber numBlocks = RelationGetNumberOfBlocks(heap);
-        uint32_t    estimated_row_count = 0;
+        BlockNumber numBlocks;
+
+        // Populate numBlocks correctly by explicitly handling each possible value forkNum can take
+        switch(forkNum) {
+            case MAIN_FORKNUM:
+                numBlocks = RelationGetNumberOfBlocks(heap);
+                break;
+            default:  // should be case INIT_FORKNUM, but set to default to keep compiler quiet
+                numBlocks = RelationGetNumberOfBlocksInFork(index, forkNum);
```

Comment:
No -- RelationGetNumberOfBlocks is a preprocessor macro that calls RelationGetNumberOfBlocksInFork and passes in MAIN_FORKNUM as the second argument, so I found it necessary to call RelationGetNumberOfBlocks and manually pass in forkNum (which would store INIT_FORKNUM in this case). In hindsight, I could have just hardcoded the second argument to INIT_FORKNUM as well to make my intention more explicit (so that I'd call RelationGetNumberOfBlocksInFork(index, forkNum) instead). Would that be better?

Reply:
So, the whole `switch` statement could just be replaced with `numBlocks = RelationGetNumberOfBlocksInFork(index, forkNum);`, no?

Reply:
Ah, you're right! That's much, much cleaner. I'll have the fixes made and changes pushed!

Reply:
Hold on a minute, forget what I said earlier. The switch statement can't be replaced with `numBlocks = RelationGetNumberOfBlocksInFork(index, forkNum);` -- when `forkNum == MAIN_FORKNUM`, the relation we pass in is `heap`, but when `forkNum == INIT_FORKNUM`, the relation we pass in instead is `index `(`heap `references NULL).